spec/abci++/abci++_methods_002_draft.md

```diff
@@ -0,0 +1,843 @@
+    |----------------------|---------------------------------------------|-------------------------------------------------------------------------------------------------------------------|--------------|
+    | hash                 | bytes                                       | The block header's hash. Present for convenience (can be derived from the block header).                          | 1            |
+    | header               | [Header](../core/data_structures.md#header) | The block header.                                                                                                 | 2            |
+    | tx                   | repeated bytes                              | List of transactions committed as part of the block.                                                              | 3            |
+    | byzantine_validators | repeated [Evidence](#evidence)              | List of evidence of validators that acted maliciously.                                                            | 4            |
+    | last_commit_info     | [LastCommitInfo](#lastcommitinfo)           | Info about the last commit, including the round, and the list of validators and which ones signed the last block. | 5            |
+
+* **Response**:
+
+    | Name                    | Type                                                        | Description                                                                      | Field Number |
+    |-------------------------|-------------------------------------------------------------|----------------------------------------------------------------------------------|--------------|
+    | block_events            | repeated [Event](abci++_basic_concepts_002_draft.md#events) | Type & Key-Value events for indexing                                             | 1            |
+    | tx_result               | repeated [DeliverTxResult](#delivertxresult)                | List of structures containing the data resulting from executing the transactions | 2            |
+    | validator_updates       | repeated [ValidatorUpdate](#validatorupdate)                | Changes to validator set (set voting power to 0 to remove).                      | 3            |
+    | consensus_param_updates | [ConsensusParams](#consensusparams)                         | Changes to consensus-critical gas, size, and other parameters.                   | 4            |
+    | app_hash                | bytes                                                       | The Merkle root hash of the application state.                                   | 6            |
+    | retain_height           | int64                                                       | Blocks below this height may be removed. Defaults to `0` (retain all).           | 7            |
```

Comment:
what about adding a field here to implement https://github.com/tendermint/tendermint/issues/1909